File: VB/FilterReportSolution.Module/DatabaseUpdate/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.Reports

Namespace FilterReportSolution.Module.DatabaseUpdate
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()

			If ObjectSpace.FindObject(Of Note)(Nothing) Is Nothing Then
				For i As Integer = 0 To 9999
					Dim n As Note = ObjectSpace.CreateObject(Of Note)()
					n.Text = String.Format("sample text {0}", i + 1)
					n.Author = String.Format("author{0}", i Mod 5 + 1)
				Next i
			End If

			CreateReport("TestReport")
		End Sub
		Private Sub CreateReport(ByVal reportName As String)
			Dim reportdata As ReportData = ObjectSpace.FindObject(Of ReportData)(New BinaryOperator("Name", reportName))
			If reportdata Is Nothing Then
				reportdata = ObjectSpace.CreateObject(Of ReportData)()
				Dim rep As New XafReport()
				rep.ObjectSpace = ObjectSpace
				rep.LoadLayout(Me.GetType().Assembly.GetManifestResourceStream(Me.GetType().Namespace + "." & reportName & ".repx"))
				rep.ReportName = reportName
				reportdata.SaveXtraReport(rep)
				reportdata.IsInplaceReport = True
				reportdata.Save()
			End If
		End Sub

	End Class
End Namespace

File: CS/FilterReportSolution.Module/DatabaseUpdate/Updater.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Security;
using DevExpress.ExpressApp.Reports;

namespace FilterReportSolution.Module.DatabaseUpdate {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();

            if (ObjectSpace.FindObject<Note>(null) == null) {
                for (int i = 0; i < 10000; i++) {
                    Note n = ObjectSpace.CreateObject<Note>();
                    n.Text = string.Format("sample text {0}", i + 1);
                    n.Author = string.Format("author{0}", i % 5 + 1);
                }
            }

            CreateReport("TestReport");
        }
        private void CreateReport(string reportName) {
            ReportData reportdata = ObjectSpace.FindObject<ReportData>(new BinaryOperator("Name", reportName));
            if (reportdata == null) {
                reportdata = ObjectSpace.CreateObject<ReportData>();
                XafReport rep = new XafReport();
                rep.ObjectSpace = ObjectSpace;
                rep.LoadLayout(GetType().Assembly.GetManifestResourceStream(this.GetType().Namespace + "." + reportName + ".repx"));
                rep.ReportName = reportName;
                reportdata.SaveReport(rep);
                reportdata.IsInplaceReport = true;
                reportdata.Save();
            }
        }

    }
}
